testing/upload_validation.ts

```diff
@@ -864,7 +864,12 @@ const syncTableSchema = z.union([genericDynamicSyncTableSchema, genericSyncTable
 const unrefinedPackVersionMetadataSchema = zodCompleteObject<PackVersionMetadata>({
   version: z
     .string()
-    .regex(/^\d+(\.\d+){0,2}$/, 'Pack versions must use semantic versioning, e.g. "1", "1.0" or "1.0.0".'),
+    .regex(/^\d+(\.\d+){0,2}$/, 'Pack versions must use semantic versioning, e.g. "1", "1.0" or "1.0.0".')
+    .refine(
+      // Version numbers must not be bigger than a postgres integer.
+      version => version.split('.').filter(part => Number(part) > 2_147_483_647).length === 0,
```

Comment:
oh just curious what is the `_` syntax in the number?

Reply:
https://2ality.com/2018/02/numeric-separators.html 

Fun fact, the guy who got that included into JS is a friend of mine :)